src/main/java/ch/njol/skript/effects/EffDetonate.java

```diff
@@ -0,0 +1,56 @@
+package ch.njol.skript.effects;
+
+import org.bukkit.entity.*;
+import org.bukkit.entity.minecart.ExplosiveMinecart;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+
+@Name("Detonate Entity")
+@Description("Automatically detonates an entity.\n Accepted entities are 'Fireworks, TNT Minecarts and WindCharges")
+@Examples("detonate last launched firework")
+@Since("INSERT VERSION")
+public class EffDetonate extends Effect {
+	static {
+		Skript.registerEffect(EffDetonate.class, "detonate [a[n]|the] %entities%");
+	}
+
+	private Expression<Entity> entities;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		this.entities = (Expression<Entity>) exprs[0];
+ 		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		for (Object entity : entities.getArray(event)) {
+			if (entity instanceof Firework) {
+				((Firework) entity).detonate();
+			}
+			if (entity instanceof WindCharge) {
```

Comment:
Add a boolean static check for the wind charge class existing, and make this an else/if tree. If it's a firework then it's not a minecart or a wind charge so we don't need to check everything else too. You should be able to check the static flag && the instance-of without causing a problem.
Also, when Skript is updated to Java whatever you'll be able to use pattern variables for this too.